src/adapter/src/error.rs

```diff
@@ -662,6 +672,13 @@ impl fmt::Display for AdapterError {
             AdapterError::MultiTableWriteTransaction => {
                 f.write_str("write transactions only support writes to a single table")
             }
+            AdapterError::DDLOnlyTransaction => f.write_str(
+                "transactions which modify objects are restricted to just modifying objects",
+            ),
+            AdapterError::DDLTransactionRace => {
+                f.write_str("object state changed while transaction was in progress")
+            }
+            AdapterError::DDLTransactionDryRun => f.write_str("ddl transaction dry run"),
```

Comment:
note: I thought about using `unreachable!(...)` here since this should never be displayed to the user, but wasn't totally sure that's a good idea.

Reply:
`soft_assert` might be a good choice for that?

Reply:
Yeah 100%, added `soft_panic_or_log!`